lib/src/scalafmt/java/com/diffplug/spotless/glue/scalafmt/ScalafmtFormatterFunc.java

```diff
@@ -0,0 +1,41 @@
+package com.diffplug.spotless.glue.scalafmt;
+
+import com.diffplug.spotless.FileSignature;
+import com.diffplug.spotless.FormatterFunc;
+
+import org.scalafmt.Scalafmt;
+import org.scalafmt.config.ScalafmtConfig;
+import org.scalafmt.config.ScalafmtConfig$;
+
+import scala.collection.immutable.Set$;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+public class ScalafmtFormatterFunc implements FormatterFunc {
+	private final FileSignature configSignature;
+
+	public ScalafmtFormatterFunc(FileSignature configSignature) {
+		this.configSignature = configSignature;
+	}
+
+	@Override
+	public String apply(String input) throws Exception {
+		ScalafmtConfig config;
+		if (configSignature.files().isEmpty()) {
+			// Note that reflection is used here only because Scalafmt has a method called
```

Comment:
One way around this may be that instead of defining `ScalafmtFormatterFunc` as a Java source file, instead we can define it as a Scala source file (which with a basic class Java has no issues calling). Not sure if this is possible or even desirable

Reply:
I'd be okay putting Scala into our codebase if it was really useful for some task, but this is small enough that I'd prefer the reflection. Note the little refactor I did in a75877c.